File: Java/program/PetShop.java
public class PetShop {
    protected String id, namaProduk;
    protected double harga;
    protected int stok;

    public PetShop(String id, String namaProduk, double harga, int stok) {
        this.id = id;
        this.namaProduk = namaProduk;
        this.harga = harga;
        this.stok = stok;
    }

    public String[] display() {
        return new String[] { id, namaProduk, String.valueOf(harga), String.valueOf(stok) };
    }

    public void updateData(String namaProduk, double harga, int stok) {
        this.namaProduk = namaProduk;
        this.harga = harga;
        this.stok = stok;
    }
}

class Aksesoris extends PetShop {
    protected String jenis, bahan, warna;

    public Aksesoris(String id, String namaProduk, double harga, int stok, String jenis, String bahan, String warna) {
        super(id, namaProduk, harga, stok);
        this.jenis = jenis;
        this.bahan = bahan;
        this.warna = warna;
    }

    @Override
    public String[] display() {
        return new String[] { id, namaProduk, String.valueOf(harga), String.valueOf(stok), jenis, bahan, warna };
    }
}

class Baju extends Aksesoris {
    private String untuk, size, merk;

    public Baju(String id, String namaProduk, double harga, int stok, String jenis, String bahan, String warna,
            String untuk, String size, String merk) {
        super(id, namaProduk, harga, stok, jenis, bahan, warna);
        this.untuk = untuk;
        this.size = size;
        this.merk = merk;
    }

    @Override
    public String[] display() {
        return new String[] { id, namaProduk, String.valueOf(harga), String.valueOf(stok), jenis, bahan, warna, untuk,
                size, merk };
    }

    public void updateBaju(String namaProduk, double harga, int stok, String jenis, String bahan, String warna,
            String untuk, String size, String merk) {
        updateData(namaProduk, harga, stok);
        this.jenis = jenis;
        this.bahan = bahan;
        this.warna = warna;
        this.untuk = untuk;
        this.size = size;
        this.merk = merk;
    }
}

File: Python/program/PetShop.py
class PetShop:
    def __init__(self, id, nama_produk, harga_produk, stok_produk):
        self.id = id
        self.nama_produk = nama_produk
        self.harga_produk = harga_produk
        self.stok_produk = stok_produk

    def display(self):
        return [self.id, self.nama_produk, self.harga_produk, self.stok_produk]

    def update_data(self, nama, harga, stok):
        self.nama_produk = nama
        self.harga_produk = harga
        self.stok_produk = stok


class Aksesoris(PetShop):
    def __init__(self, id, nama, harga, stok, jenis, bahan, warna):
        super().__init__(id, nama, harga, stok)
        self.jenis = jenis
        self.bahan = bahan
        self.warna = warna

    def display(self):
        return super().display() + [self.jenis, self.bahan, self.warna]


class Baju(Aksesoris):
    def __init__(self, id, nama, harga, stok, jenis, bahan, warna, untuk, size, merk):
        super().__init__(id, nama, harga, stok, jenis, bahan, warna)
        self.untuk = untuk
        self.size = size
        self.merk = merk

    def display(self):
        return super().display() + [self.untuk, self.size, self.merk]

    def update_baju(self, nama, harga, stok, jenis, bahan, warna, untuk, size, merk):
        self.update_data(nama, harga, stok)
        self.jenis = jenis
        self.bahan = bahan
        self.warna = warna
        self.untuk = untuk
        self.size = size
        self.merk = merk
